applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp

```diff
@@ -8,20 +8,173 @@
+
+    // Verify that the variables are correctly initialized
+    KRATOS_CHECK_VARIABLE_KEY(DISPLACEMENT)
+    KRATOS_CHECK_VARIABLE_KEY(VELOCITY)
+    KRATOS_CHECK_VARIABLE_KEY(ACCELERATION)
+    KRATOS_CHECK_VARIABLE_KEY(DENSITY)
+    KRATOS_CHECK_VARIABLE_KEY(VOLUME_ACCELERATION)
+    KRATOS_CHECK_VARIABLE_KEY(THICKNESS)
+
+    // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
+    for ( IndexType i = 0; i < number_of_nodes; i++ ) {
+        const NodeType &rnode = r_geometry[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISPLACEMENT,rnode)
+
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_X, rnode)
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Y, rnode)
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Z, rnode)
+    }
+
+    // Verify that the constitutive law exists
+    KRATOS_ERROR_IF_NOT(r_properties.Has( CONSTITUTIVE_LAW )) << "Constitutive law not provided for property " << r_properties.Id() << std::endl;
+
+    // Verify that the constitutive law has the correct dimension
+    const SizeType strain_size = r_properties.GetValue( CONSTITUTIVE_LAW )->GetStrainSize();
+    if ( dimension == 2 ) {
+        KRATOS_ERROR_IF( strain_size < 3 || strain_size > 4) << "Wrong constitutive law used. This is a 2D element! expected strain size is 3 or 4 (el id = ) " << rElement.Id() << std::endl;
+    } else {
+        KRATOS_ERROR_IF_NOT(strain_size == 6) << "Wrong constitutive law used. This is a 3D element! expected strain size is 6 (el id = ) "<<  rElement.Id() << std::endl;
+    }
+    
+    // Check constitutive law
+    if ( rConstitutiveLaws.size() > 0 ) {
+        return rConstitutiveLaws[0]->Check( r_properties, r_geometry, rCurrentProcessInfo );
+    }
+
+    return 0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeEquivalentF(
+    const Element& rElement,
+    const Vector& rStrainTensor,
+    Matrix& rF
+    )
+{
+    const auto& r_geometry = rElement.GetGeometry();
+    const SizeType dimension = r_geometry.WorkingSpaceDimension();
+
+    if(dimension == 2) {
+        rF(0,0) = 1.0+rStrainTensor(0);
+        rF(0,1) = 0.5*rStrainTensor(2);
+        rF(1,0) = 0.5*rStrainTensor(2);
+        rF(1,1) = 1.0+rStrainTensor(1);
+    } else {
+        rF(0,0) = 1.0+rStrainTensor(0);
+        rF(0,1) = 0.5*rStrainTensor(3);
+        rF(0,2) = 0.5*rStrainTensor(5);
+        rF(1,0) = 0.5*rStrainTensor(3);
+        rF(1,1) = 1.0+rStrainTensor(1);
+        rF(1,2) = 0.5*rStrainTensor(4);
+        rF(2,0) = 0.5*rStrainTensor(5);
+        rF(2,1) = 0.5*rStrainTensor(4);
+        rF(2,2) = 1.0+rStrainTensor(2);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void CalculateB(
+    const Element& rElement,
```

Comment:
all of these methods should be templated in the matrix types. this is important to also allow using them when the mattix type is static